### PR TITLE
REGR: window ewm slowdown

### LINKED ISSUE #42333: REGR: rolling.EWMMethods.time_ewm
This seems to have resulted in a performance regression for the existing ASV benchmark: https://pandas.pydata.org/speed/pandas/#rolling.EWMMethods.time_ewm?python=3.8&Cython=0.29.21&p-constructor='DataFrame'&p-constructor='Series'&p-window=1000&p-dtype='float'&p-method='mean'&commits=ab687aec-f066cda6

_Originally posted by @jorisvandenbossche in https://github.com/pandas-dev/pandas/issues/40072#issuecomment-787456038_

### PR DESCRIPTION
**Pull request by mzeitlin11**
- [x] closes #42333
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry (will move once 1.3.3 notes merged)

```
-     3.22±0.09ms       1.29±0.3ms     0.40  rolling.EWMMethods.time_ewm('DataFrame', 10, 'int', 'mean')
-      2.92±0.1ms       1.08±0.1ms     0.37  rolling.EWMMethods.time_ewm('DataFrame', 1000, 'int', 'mean')
-      2.68±0.1ms        959±100μs     0.36  rolling.EWMMethods.time_ewm('DataFrame', 1000, 'float', 'mean')
-      3.00±0.2ms      1.02±0.03ms     0.34  rolling.EWMMethods.time_ewm('DataFrame', 10, 'float', 'mean')
-      2.78±0.1ms        850±100μs     0.31  rolling.EWMMethods.time_ewm('Series', 10, 'int', 'mean')
-      2.34±0.2ms         683±30μs     0.29  rolling.EWMMethods.time_ewm('Series', 10, 'float', 'mean')
-     2.74±0.06ms         739±60μs     0.27  rolling.EWMMethods.time_ewm('Series', 1000, 'float', 'mean')
-      2.55±0.3ms         681±40μs     0.27  rolling.EWMMethods.time_ewm('Series', 1000, 'int', 'mean')
```

### CODE REVIEW

**Comment by jbrockmendel** [comment]
nice speedup, cc @mroeschke

File: doc/source/whatsnew/v1.3.2.rst [hunk 1]
```diff
@@ -15,6 +15,7 @@ including other versions of pandas.
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
 - Performance regression in :meth:`DataFrame.isin` and :meth:`Series.isin` for nullable data types (:issue:`42714`)
+- Performance regression in :meth:`core.window.ewm.ExponentialMovingWindow.mean` (:issue:`42333`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.2.rst:18]: move to 1.3.3

**Review by mroeschke — APPROVED** [review]
Thanks! LGTM

File: pandas/_libs/window/aggregations.pyx [hunk 2]
```diff
@@ -1489,7 +1489,7 @@ def roll_weighted_var(const float64_t[:] values, const float64_t[:] weights,
 
 def ewma(const float64_t[:] vals, const int64_t[:] start, const int64_t[:] end,
          int minp, float64_t com, bint adjust, bint ignore_na,
-         const float64_t[:] deltas) -> np.ndarray:
+         const float64_t[:] deltas=None) -> np.ndarray:
```
Comment (jreback) [inline pandas/_libs/window/aggregations.pyx:1492]: can u update the doc string

**Review by jreback — CHANGES_REQUESTED** [review]
yep lgtm

File: pandas/_libs/window/aggregations.pyx (hunk 2 above)
Comment (mzeitlin11) [reply pandas/_libs/window/aggregations.pyx:1492]: done

File: doc/source/whatsnew/v1.3.2.rst (hunk 1 above)
Comment (mzeitlin11) [reply doc/source/whatsnew/v1.3.2.rst]: done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mzeitlin11

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**